sway-fmt-v2/src/utils/map/newline.rs

```diff
@@ -182,12 +199,18 @@ fn insert_after_span(
 ) -> Result<usize, FormatterError> {
     let iter = newline_sequences_to_insert.iter();
     let mut sequence_string = String::new();
+    let mut newlines_to_skip =
+        find_already_present_extra_newlines(from.end, formatted_code.to_string());
     for newline_sequence in iter {
-        write!(
-            sequence_string,
-            "{}",
-            format_newline_sequnce(newline_sequence, threshold)
-        )?;
+        if newlines_to_skip != 0 {
+            newlines_to_skip -= 1;
+        } else {
+            write!(
+                sequence_string,
+                "{}",
+                format_newline_sequnce(newline_sequence, threshold)
+            )?;
+        }
     }
```

Comment:
It looks like this could also be written as:

```rust
for newline_sequence in iter.skip(newlines_to_skip) {
    write!(
        sequence_string,
        "{}",
         format_newline_sequnce(newline_sequence, threshold),
    )?;
}
```

Reply:
Actually, I first wrote it like this but realized that a newline sequence can have multiple newlines in it.

For example

```rust
script;

fn main() {
    let number = 10;
    let number5 = 10;
}
```

would produce the following newline sequences:

1. newline_sequence { start: .., end: .. length: 2}
2. newline_sequence { start: .., end: .. length: 1}
3. newline_sequence { start: .., end: .. length: 1}

(I did not took those from a debug output so exact field name etc can change but the logic is like this)

Reply:
Hmm I'm not quite sure I'm following here - I think your implementation in this PR and my suggestion look to be semantically equivalent?

```rust
    for newline_sequence in iter {
        if newlines_to_skip != 0 {
            newlines_to_skip -= 1;
        } else {
            write!(
                sequence_string,
                "{}",
                format_newline_sequnce(newline_sequence, threshold)
            )?;
        }
    }
```
^ Here, for each value of `newlines_to_skip` that is greater than `0`, we skip writing that `newline_sequence` element. It looks to me like this is the same as skipping the first `newlines_to_skip` number of elements. A more concise way of writing this is using the `skip` adapter like so:

```rust
for newline_sequence in iter.skip(newlines_to_skip) {
    write!(
        sequence_string,
        "{}",
         format_newline_sequnce(newline_sequence, threshold),
    )?;
}
```

Let me know if I'm still missing something here! Maybe the intended logic is a little different?

Reply:
Ahh, yeah my bad. While trying to understand the problem that popped up during testing (the one that I mentioned through slack) I thought this was the problem than realized it was not the case 🤦